### ISSUE TITLE
Auto-discover cluster nodes

### ISSUE THREAD

**Issue by gabmontes** [bloq/memproxyjs#17]
Replace `memcached` with [`memcached-elasticache`](https://www.npmjs.com/package/memcached-elasticache) to be able to use the configuration endpoint of the cluster instead of an array of nodes. This way, the driver will automatically discover/add/remove nodes as the cluster configuration changes.

**Comment by jgarzik** [comment]
_"redis and memcached [are not good] for authentication & encryption of transport TCP connection.   This has a large trickle-down architecture impact.
One solution is the network route:   Create an VPC, which is very difficult with Lambda @ Edge deployment (M regions -> 1 VPC == many tunnels).  then you must encrypt that network end-to-end (somehow).  Besides difficulty, that hurts our @Edge goal.
The other solution is to wrap nginx authentication & encryption, making the easiest path HTTP protocol.  thus memproxyjs."_

tl;dr We need our own configuration endpoint, probably a simple lambda that queries the target group members.
